jukebox/src/main/webapp/js/dbScript.js

```diff
@@ -33,7 +33,8 @@ async function addSongToDB(song_ID, title) {
             });
         } else {
             db.collection("songs").doc(song_ID).set( {
-                SONG_NAME: title
+                SONG_NAME: title,
```

Comment:
I think we can simplify this logic a bit so that we don't need to keep doing duplicate calls here. Let me know if anything I'm saying doesn't work here.

If we do an update https://firebase.google.com/docs/reference/js/firebase.database.Reference#update on the song first, we can ensure it's in Firebase first before doing proceeding. If it already exists, nothing will change (I would imagine it would be able to determine that no actions need to be performed) and if unset, it'll add it to Firebase. 

From there, we can use the callback to add it to the queue. This way, we only need to add the code for the addition once. Does that work, or am I missing anything?

Reply:
Yeah, that makes sense. I'm not entirely caught up on how the current indexing situation works, so I'll look into it and try to remove duplicate calls.

Reply:
I think I was able to make the logic simpler by taking into account suggestions you made. The code for adding a song to the queue now only occurs once, however I didn't end up using the update function (I think the logic is the same as you've laid out, but without the update function).

Right now we check if the song exists through a read rather than an update, and then we do a write either way. I think the firebase model works so that reads are actually cheaper than updates, so the method of reading to check if the song exists as opposed to calling an update to check is less costly.

I've also made the queue index equal to the time at which the server receives the write call, which would reduce the need for 1 read and 1 update operation while using the same queue ordering. Let me know if what I've written needs changing or if I misunderstood something. Thanks!

Here's what the function looks like now:

```
async function addSongToDB(song_ID, title, thumbnailUrl) {
    const room_ID = getPin();
    var currQueue = db.collection('rooms').doc(room_ID).collection("queue");
    
    db.collection('songs').doc(song_ID)
    .get()
    .then(function(querySnapshot) {
        if(!querySnapshot.exists){
            db.collection("songs").doc(song_ID).set( {
                SONG_NAME: title,
                THUMBNAIL_URL: thumbnailUrl
            });
        }
        currQueue.doc(song_ID).set({
            SONG_ID: song_ID,
            ROOM_ID: room_ID,
            QUEUE_INDEX: firebase.firestore.FieldValue.serverTimestamp()
        });
    })
    .catch(function(error) {
        console.log("Error getting documents: ", error);
    })
}
```

Reply:
That solution looks good to me.